gw_eccentricity/eccDefinition.py

```diff
@@ -1749,6 +2102,29 @@ def measure_ecc(self, tref_in=None, fref_in=None):
         # return measured eccentricity, mean anomaly and reference time or
         # frequency where these are measured.
         return self.make_return_dict_for_eccentricity_and_mean_anomaly()
+    
+    def check_omega_gw_extrema_interpolants(self):
+        """
+        check monotonicity of the omega_gw extrema interpolants.
+        """
+        # Verify the monotonicity of the first derivative of the omega_gw
+        # interpolant with spline.
+        if self.extra_kwargs["omega_gw_extrema_interpolation_method"] \
+            == "spline":
```

Comment:
Why only for spline? If it's because you already do this test for rational_fits, should this check happen within the method that builds the spline? Otherwise, the code seems confusing. 

Similarly, I'm not a fan of if conditions like this
https://github.com/vijayvarma392/gw_eccentricity/blob/b8872a1e4bb44f74287583007d995fb3003eaeae/gw_eccentricity/eccDefinition.py#L1626
Is there a way to just call this test from within the function that does that rational fit?

Reply:
Done